src/Spectre.Data/Datasets/BasicTextDataset.cs

```diff
@@ -0,0 +1,307 @@
+ * file containing formatted data.
+ * 
+   Copyright 2017 Dariusz Kuchta
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.IO;
+using Spectre.Data.Structures;
+
+namespace Spectre.Data.Datasets
+{
+    public class BasicTextDataset : IDataset
+    {
+
+        //TODO: Because of not-so failsafe constructors, make the Datasets 
+        //TODO: be eventually spitted out by some factory class!!!
+
+        #region Fields
+
+        /// <summary>
+        /// Metadata of the dataset.
+        /// </summary>
+        private Metadata _metadata;
+
+        /// <summary>
+        /// Container for storing spacial coordinates for every loaded spectrum.
+        /// </summary>
+        private List<SpatialCoordinates> _spatialCoordinates;
+
+        /// <summary>
+        /// Array of m/z values for all the spectras.
+        /// </summary>
+        private double[] _mz;
+
+        /// <summary>
+        /// Container for storing intensity values for every loaded spectrum.
+        /// </summary>
+        private List<double[]> _intensity;
+
+        #endregion
+
+        #region Constructors
+
+        /// <summary>
+        /// Constructor with file initialization.
+        /// </summary>
+        /// <param name="textFilePath">Path to the text file.</param>
+        public BasicTextDataset(string textFilePath)
+        {
+            CreateFromFile(textFilePath);
+        }
+        /// <summary>
+        /// Constructor with raw value initialization.
+        /// </summary>
+        /// <param name="mz">Array of m/z values.</param>
+        /// <param name="data">Array of intensity values.</param>
+        public BasicTextDataset(double[] mz, double[,] data)
+        {
+            CreateFromRawData(mz, data);
+        }
+
+        #endregion
+
+        #region IDataset
+
+        /// <summary>
+        /// See <see cref="IDataset"/> for description.
+        /// </summary>
+        public Metadata Metadata
+        {
+            get { return _metadata; }
+            private set { _metadata = value; }
+        }
+        /// <summary>
+        /// See <see cref="IDataset"/> for description.
+        /// </summary>
+        public IEnumerable<SpatialCoordinates> SpacialCoordinates
+        {
+            get { return _spatialCoordinates; }
+            private set { _spatialCoordinates = value as List<SpatialCoordinates>; }
```

Comment:
No type check is done - if `value` is not `List<SpatialCoordinates>`, `null` will be assigned. Such check can be maintained only during assignment, outside (or an Exception should be thrown).
On the other hand, you should be able to [dump arbitrary `IEnumerable` to `List`](http://stackoverflow.com/a/7617783/3067956)...